### PR TITLE
docs: make constructor options linkable

### PR DESCRIPTION
**Pull request by alanshaw**
This PR changes the constructor options into headings so that people can link the relevant options. I have needed this multiple times but haven't been able to give people a link to the correct part of the page.

### CODE REVIEW

**Comment by PascalPrecht** [comment]
Sweet! ⚡️

**Comment by Mr0grog** [comment]
Will try and get a look at this later today.

File: README.md [hunk 1]
```diff
@@ -202,56 +202,122 @@ const node = new IPFS([options])
...
-        - `peerDiscovery` (object):
-            - `[PeerDiscovery.tag]` (object): configuration for a peer discovery module
-                - `enabled` (boolean): whether this module is enabled or disabled
-                - `[custom config]` (any): other keys are specific to the module
+| Type | Default |
+|------|---------|
+| object | [defaults](https://github.com/libp2p/js-libp2p-connection-manager#create-a-connectionmanager) |
 
-- `connectionManager` (object): Configure the libp2p connection manager, see the [documentation for available options](https://github.com/libp2p/js-libp2p-connection-manager#create-a-connectionmanager).
+Configure the libp2p connection manager.
```
[the remark this answers is not in the document]
Comment (alanshaw) [reply README.md]: gah, no

[the remark this answers is not in the document]
Comment (alanshaw) [reply README.md]: Wait, it is linked from the table above instead

**Pull request merged**